Homework-3-PostgreSQL-Layered-Architecture/src/api/users.ts

```diff
@@ -0,0 +1,102 @@
+import express, { Request, Response, NextFunction } from 'express';
+import {
+    ValidatedRequest,
+    createValidator
+} from 'express-joi-validation';
+import { Container } from "typedi";
+import { IRequestBodySchema, reqBodySchema } from "../schemas/userSchema";
+import { UserService } from "../services/userService";
+import { BaseUserDTO, UserDTO } from "../types/userDTO";
+import { UserFilter } from "../types/filters";
+
+export const usersRouter = express.Router();
+const validator = createValidator({ passError: true });
+const userService = Container.get(UserService);
+
+const getUsers = async (req: Request, res: Response, next: NextFunction) => {
+  try {
+    const filters: UserFilter = req.query;
```

Comment:
It'd be nice to validate input parameters. Currently I can pass anything in `limit` and will get 500 error.

Reply:
Will be done in next homework